### DISCUSSION TITLE
Can sembast's db file readed at native side?

### DISCUSSION THREAD

**Discussion by lanistor** [tekartik/sembast.dart#381] [Q&A] [answered]
IOS widgets cannot start flutter, so the data on widgets cannot update. If sembast's db file can readed on iOS Native side, like by sqlite, This problem can be resolved.

**Comment by alextekartik** [comment, marked as answer]
Sembast db could be read on the native side. If you use sembast (io), it is a simple text file of json lines. The format is described here: https://github.com/tekartik/sembast.dart/blob/master/sembast/doc/storage_format.md

So you will likely have to implement your own native parser and loader.

If you use sembast_sqflite, the format is similar but on a sqlite database that you could open from the iOS side.

Of course the proprietary internal format could change (although it has never change for 7 years!)
